src/main/java/org/logstash/dissect/search/DelimiterSearchStrategy.java

```diff
@@ -0,0 +1,5 @@
+package org.logstash.dissect.search;
+
+public interface DelimiterSearchStrategy {
```

Comment:
FYI, the only reason for operating on bytes is that the JRuby RubyString, being mutable, is backed by a custom ByteList class - there is no Java String value as state - these bytes (in UTF8, because its a rule in LS that any strings added to an Event must be converted to UTF8 before) are readily available.

I changed the Delimiter create to this:
```java
    public static Delimiter create(String delim) {
        byte[] bytes = delim.getBytes();
        switch (bytes.length) {
            case 0 :
                return new Delimiter(delim, bytes, ZeroByteLocator.INSTANCE);
            case 1 :
                return new Delimiter(delim, bytes, SingleByteLocator.INSTANCE);
            case 2 :
                return new Delimiter(delim, bytes, DoubleByteLocator.INSTANCE);
            default :
                return new Delimiter(delim, bytes, MultiByteLocator.INSTANCE);
        }
    }
```